modules/vislab/rendering/renderabledistancelabel.cpp

```diff
@@ -0,0 +1,107 @@
+ * permit persons to whom the Software is furnished to do so, subject to the following   *
+ * conditions:                                                                           *
+ *                                                                                       *
+ * The above copyright notice and this permission notice shall be included in all copies *
+ * or substantial portions of the Software.                                              *
+ *                                                                                       *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,   *
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A         *
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT    *
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF  *
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include <modules/vislab/rendering/renderabledistancelabel.h>
+
+#include <modules/base/rendering/renderablenodeline.h>
+#include <openspace/documentation/documentation.h>
+#include <openspace/documentation/verifier.h>
+#include <openspace/engine/globals.h>
+#include <openspace/rendering/renderengine.h>
+#include <openspace/scene/scene.h>
+#include <string>
+
+namespace {
+    constexpr const char* _loggerCat = "RenderableDistanceLabel";
+
+    constexpr openspace::properties::Property::PropertyInfo NodeLineInfo = {
+        "NodeLine",
+        "Node Line",
+        "Property to track a nodeline. When tracking the label text will be updating the distance "
+        "from the nodeline start and end. "
+    };
+}
+
+namespace openspace {
+
+documentation::Documentation RenderableDistanceLabel::Documentation() {
+    using namespace documentation;
+    return {
+        "Renderable Distance Label",
+        "vislab_renderable_distance_label",
+        {
+            {
+                NodeLineInfo.identifier,
+                new StringVerifier,
+                Optional::No,
+                NodeLineInfo.description
+            },
+        }
+    };
+}
+
+RenderableDistanceLabel::RenderableDistanceLabel(const ghoul::Dictionary& dictionary)
+    : RenderableLabels(dictionary)
+    , _nodeLine(NodeLineInfo)
+{
+    documentation::testSpecificationAndThrow(
+        Documentation(),
+        dictionary,
+        "RenderableDistanceLabel"
+    );
+
+    if (dictionary.hasKey(NodeLineInfo.identifier)) {
+        _nodeLine = dictionary.value<std::string>(NodeLineInfo.identifier);
+        addProperty(_nodeLine);
+    }
+}
+
+void RenderableDistanceLabel::update(const UpdateData& data) {
+
+    if (global::renderEngine.scene()->sceneGraphNode(_nodeLine)) {
```

Comment:
Might as well assign to nodelineNode and check it for nullptr, since we're getting it on the row below.